### PR TITLE
Fix 7055 one of with scalar value and string

### LINKED ISSUE #7055: Format schema with oneOf doesn't work correctly
```ts
const schema = {
  type: 'object',
  properties: {
    from: {
      format: 'address',
    },
    to: {
      oneOf: [{ format: 'string' }, { type: 'null' }],
    },
  },
};

const data = {
  from: '0x7ed0e85b8e1e925600b4373e6d108f34ab38a401',
  to: 123,
};

const res = format(schema, data);

// res { from: '0x7ed0e85b8e1e925600b4373e6d108f34ab38a401', to: 123 }
```

**Actual Behaviour:**
`to` is not formatted

**Expected Behaviour:**
`to` should be formatted to a string

### PR DESCRIPTION
**Pull request by mmyyrroonn**
## Description

Fix #7055

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-utils/test/unit/formatter.test.ts [hunk 1]
```diff
@@ -479,6 +479,26 @@ describe('formatter', () => {
...
+						format({ format: 'string' }, 123, {
+							number: FMT_NUMBER.STR,
+							bytes: FMT_BYTES.HEX,
+						}),
+					).toBe('123');
+				});
+
+				it('should format string for 0x1234', () => {
+					expect(
+						format({ format: 'string' }, 0x1234, {
```
Comment (luu-alex) [inline packages/web3-utils/test/unit/formatter.test.ts]:
```suggestion
				it('should format string for 1234, () => {
					expect(
						format({ format: 'string' }, 1234, {
```
We don't actually test using hex, formatting with numbers imo would be better

File: packages/web3-utils/test/unit/formatter.test.ts [hunk 2]
```diff
@@ -827,6 +847,32 @@ describe('formatter', () => {
 
 				expect(result).toEqual(expected);
 			});
+
+			it('should format object with oneOf', () => {
```
Comment (luu-alex) [inline packages/web3-utils/test/unit/formatter.test.ts:842]: can you also create a test using oneOf when the `to` property is undefined?

**Comment by luu-alex** [comment]
Thanks for submitting the PR! I have some feedback, but overall it looks good :) thanks alot. Also update the [changelog.](https://github.com/web3/web3.js/blob/4.x/packages/web3-utils/CHANGELOG.md)
Add this under unreleased as a fix.

**Comment by mmyyrroonn** [comment]
> Thanks for submitting the PR! I have some feedback, but overall it looks good :) thanks alot. Also update the [changelog.](https://github.com/web3/web3.js/blob/4.x/packages/web3-utils/CHANGELOG.md) Add this under unreleased as a fix.

Sure. Let me add more tests and update the README

File: packages/web3-utils/test/unit/formatter.test.ts (hunk 2 above)
Comment (mmyyrroonn) [reply packages/web3-utils/test/unit/formatter.test.ts:842]: Hi, Could you give me an example of this one?  Do you mean something like this?
```
const data ={
    from: '0x7ed0e85b8e1e925600b4373e6d108f34ab38a401'
};

const result = { from: '0x7ed0e85b8e1e925600b4373e6d108f34ab38a401'};
```

Comment (luu-alex) [reply packages/web3-utils/test/unit/formatter.test.ts:842]: yes

**Review by luu-alex — APPROVED** [review]

**Comment by luu-alex** [comment]
Thank you @mmyyrroonn for the PR. We will get this in by next release :)

**Review by avkos — APPROVED** [review]

**Pull request merged**